zkir/Dialect/Field/Conversions/ExtFieldToLLVM/ExtFieldToLLVM.cpp

```diff
@@ -55,14 +65,18 @@ struct ConvertExtFromCoeffs : public ConvertOpToLLVMPattern<ExtFromCoeffsOp> {
   LogicalResult
   matchAndRewrite(ExtFromCoeffsOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    auto loc = op.getLoc();
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
     auto structType = typeConverter->convertType(op.getType());
-    if (isa<QuadraticExtFieldType>(op.getType())) {
-      if (adaptor.getInput().size() != 2)
-        return op.emitOpError(
-            "expected two input types for quadratic extension field");
-      auto extFieldStruct = SimpleStructBuilder<2>::initialized(
-          rewriter, loc, structType, adaptor.getInput());
+
+    if (auto extField = dyn_cast<ExtensionFieldTypeInterface>(op.getType())) {
+      unsigned n = extField.getDegreeOverBase();
+      if (adaptor.getInput().size() != n)
+        return op.emitOpError("expected ") << n << " coefficients";
```

Comment:
I think this validation should be performed at different place and do we need this verification?? I thought current operation doesn't allow us to make this malformed value